### PR TITLE
refactor: optimization chapter

### PR DESCRIPTION
**Pull request by be-marc**

### CODE REVIEW

**Comment by be-marc** [comment]
> Keep in mind that nested resampling is a statistical procedure to estimate the predictive performance of the model trained on the full dataset.
Nested resampling is not a procedure to select optimal hyperparameters.
The resampling produces many hyperparameter configurations which should be not used to construct a final model [@Simon2007].
This means that nested resampling is an additional step after fitting a [final model](#sec-final-model).

@sumny You should add these sentences back. We have to make a connection to the final model and say that it is an additional step after tuning.

> Nested resampling is not a procedure to select optimal hyperparameters.
The resampling produces many hyperparameter configurations which should be not used to construct a final model [@Simon2007].

Many of our users get this wrong. We have to make that clear.

**Comment by RaphaelS1** [comment]
Hi @be-marc as this chapter looks like the first one to be publication-ready I'm going to make some changes directly to your PR if you don't mind so we can get it merged ASAP in a publishable form. I'll make a note of changes I make in the style/structure issues and tell people to copy this chapter as a general guide

**Comment by RaphaelS1** [comment]
@be-marc I think I've done as much as I can today. I've added a few 'TODO's for you:

1. [ ] Please add a 150-200 word abstract to the chapter
2. [ ] Please discuss amongst yourselves and add authors to the chapter (people that made major contributions to content)
3. [ ] Please add suggested exercises at the end

File: book/optimization.qmd [hunk 1]
```diff
@@ -1,173 +1,236 @@
...
+    orcid:
+    email:
+    affiliations:
+      - name: Affiliation 1
+  - name: Author 2
+    orcid:
+    email:
+    affiliations:
+      - name: Affiliation 2
+abstract: TODO (150-200 WORDS)
```
Comment (RaphaelS1) [inline book/optimization.qmd:13]: Add abstract here

File: book/optimization.qmd [hunk 2]
```diff
@@ -1,173 +1,236 @@
+---
+author:
+  - name: Author 1
```
Comment (RaphaelS1) [inline book/optimization.qmd:3]: Fill with actual authors

File: book/optimization.qmd [hunk 3]
```diff
@@ -1,173 +1,236 @@
...
+  tsk("penguins_simple"),
+  c(lrn("classif.ranger", num.trees = 1, id = "1 tree"),
+    lrn("classif.ranger", num.trees = 10, id = "10 trees"),
+    lrn("classif.ranger", num.trees = 100, id = "100 trees")),
+  rsmp("cv", folds = 3)
+)))
+```
+
+Human trial-and-error (which is essentially what we are doing above), is time-consuming, often biased, error-prone, and computationally irreproducible.
+Instead, many sophisticated HPO methods (@sec-tuner) (or 'tuners') have been developed over the last few decades for robust and efficient HPO.
```
Comment (RaphaelS1) [inline book/optimization.qmd:47]: This feels a bit out of place, needs a better connection. ALso maybe explain what evolutionary algs or Bayesian methods are? A couple examples

File: book/optimization.qmd [hunk 4]
```diff
@@ -1,173 +1,236 @@
...
+Most HPO methods are iterative and propose different configurations until some termination criterion is met, at which point the optimal configuration is then returned (@fig-optimization-loop).
```
Comment (RaphaelS1) [inline book/optimization.qmd:48]: This doesn't explain what 'fidelity levels' are, I'm still not sure from context

File: book/optimization.qmd [hunk 5]
```diff
@@ -1,173 +1,236 @@
...
+Popular, modern examples are given by algorithms based on evolutionary algorithms or Bayesian optimization methods.
```
Comment (RaphaelS1) [inline book/optimization.qmd:49]: Link to multi-fidelity is confusing, needs to be made clearer. What is 'cheap lower fidelity evaluations'?

File: book/optimization.qmd [hunk 6]
```diff
@@ -1,173 +1,236 @@
...
+Recent HPO methods often also make use of evaluating a configuration at multiple so-called fidelity levels, e.g., a neural network can be trained for an increasing number of epochs, gradient boosting can be performed for an increasing number of boosting steps and training data can always be subsampled to only include a smaller fraction of all available data.
+The general idea of multi-fidelity HPO methods is that the performance of a model obtained by using computationally cheap lower fidelity evaluations (few numbers of epochs or boosting steps, only using a small sample of all available data for training) is predictive of the performance of the model obtained using computationally expensive higher fidelity evaluations and this concept can be leveraged to make HPO more efficient (e.g., only continuing to evaluate those configurations on higher fidelities that appear to be promising).
```
Comment (RaphaelS1) [inline book/optimization.qmd:51]: What do you mean here? Is this the background for multiobjective or the definition of it?

**Review by RaphaelS1 — COMMENTED** [review]
Will make more comments tomorrow

File: book/optimization.qmd [hunk 7]
```diff
@@ -179,152 +242,334 @@ The tuner proceeds as follows:
...
-We plot the performances depending on the evaluated `cp` and `minsplit` values (see. @fig-surface).
+We plot the performances depending on the evaluated `cost` and `gamma` values (see. @fig-surface).
 
 ```{r optimization-016}
 #| label: fig-surface
-#| fig-cap: Performance depending on the evaluated cp and minsplit values.
+#| fig-cap: Performance depending on the evaluated cost and gamma values.
 
 autoplot(instance, type = "surface")
 ```
```
Comment (RaphaelS1) [inline book/optimization.qmd:331]: This plot must be fixed. Look at the legend for classif.ce, too squashed

File: book/optimization.qmd [hunk 8]
```diff
@@ -179,152 +242,334 @@ The tuner proceeds as follows:
...
 The trained model can now be used to predict new, external data.
-The tuning process can also be automated without the need to extract information on the best hyperparameter settings at the end.
-See the section on [automatic tuning](#sec-autotuner).
+The tuning process can also be automated without the need to extract information on the best hyperparameter settings at the end (@sec-autotuner).
 
 :::{.callout-warning}
 A common mistake is to report the performance estimated on the resampling sets on which the tuning was performed (`instance$result$classif.ce`) as the model's performance.
-To get statistically unbiased performance estimates for a given task, [nested resampling](#sec-model-performance) is required.
+To get statistically unbiased performance estimates for a given task, nested resampling (@sec-nested-resampling-unbiased-performance) is required.
 :::
```
Comment (RaphaelS1) [inline book/optimization.qmd:379]: Please look at the Technical chapter and make sure this doesn't overlap. Either delete content here and just refer to Technical or really simplify this. If you keep it then make sure your example shows an actual error

**Comment by RaphaelS1** [comment]
@be-marc I've made some significant changes to the structure and minor changes to content. Please look through and let me know if you have any big concerns. Personally I think the flow is more natural now but let me know your thoughts

File: book/optimization.qmd [hunk 9]
```diff
@@ -422,47 +632,86 @@ Nested resampling is not restricted to hyperparameter tuning.
...
-```{r optimization-031}
-#| echo: false
-rr$score()[, c("iteration", "classif.ce"), with = FALSE]
+```{r optimization-043}
+rr$score()[, list(iteration, classif.ce)]
 ```
 
 We see a slightly over-optimistic performance estimation on the inner resampling.
 
+If we want to examine the results even more in detail, we can also display the inner archives.
```
Comment (RaphaelS1) [inline book/optimization.qmd:658]: Please can you check this? What do you mean by lower performance? Lower CE? Is this overfitting?

**Comment by RaphaelS1** [comment]
@be-marc I've finally finished my full review. Amazing work on this chapter!! I've restructured and made some changes to wording but kept content the same. Please take a look at my changes and let me know if you have any comments/concerns. If possible I'd like to send this to the publisher on Friday

File: book/basics.qmd [hunk 10]
```diff
@@ -19,8 +19,8 @@ This model can be saved in the [learner](#learners) object and is then used to p
 `r mlr3` offers a number of different [measures](#measure) to quantify how well a model performs based on the difference between predicted and actual values.
 Usually, this [measure](#measure) is a numeric score.
 
-Splitting data into training and test sets, building a model, and evaluating it can be repeated several times, [resampling](#resampling) different training and test sets from the original data each time.
-Multiple [resampling iterations](#resampling) allow us to get a better and less biased generalizable performance estimate for a particular type of model.
+Splitting data into training and test sets, building a model, and evaluating it can be repeated several times, resampling different training and test sets from the original data each time (@sec-resampling).
+Multiple resampling iterations allow us to get a better and less biased generalizable performance estimate for a particular type of model.
```
Comment (berndbischl) [inline book/basics.qmd:23]:
```suggestion
Splitting data into training and test sets, building a model, and evaluating it can be repeated several times, resampling dsfdfgdfdfg    particular type of model.
```

**Comment by RaphaelS1** [comment]
@be-marc let me know if you need anything from me after our talk with Bernd yesterday to help get this merged

**Comment by RaphaelS1** [comment]
Comments from #453 to fix before merging:

* [ ] Split sentences into new lines
* [ ] Make code snippets less ugly
* [ ] Consider extending tuning diagram (we can redraw without mermaid to save space)
* [ ] log[1e5..] -> $log[1e5...]$ (i.e. wrap in '$')
* [ ] Change simulation example from `summary` to two `hist` (with patchwork)
